modules/core/src/test/scala/org/tessellation/infrastructure/trust/storage/TrustStorageSuite.scala

```diff
@@ -0,0 +1,388 @@
+      ordinal = Refined.unsafeApply[Long, NonNegative](delta)
+    } yield SnapshotOrdinal(ordinal)
+
+    forall(gen) { ordinal =>
+      for {
+        store <- TrustStorage.make
+
+        maybeOrdinalPublicTrust <- store.updateNext(ordinal)
+
+        expectedTrust <- store.getTrust
+      } yield
+        expect.all(
+          Order.lt(SnapshotOrdinal(ordinalTrustUpdateInterval), ordinal),
+          isEmpty(expectedTrust),
+          maybeOrdinalPublicTrust.isEmpty
+        )
+    }
+  }
+
+  test("next ordinal trust storage updates from internal storage because expected ordinal has been reached") {
+    val minOrdinalValue = ordinalTrustUpdateInterval + 1L
+    val maxOrdinalValue = ordinalTrustUpdateInterval * 2L
+
+    val gen = for {
+      trustInfo <- Gen.mapOfN(4, genPeerTrustInfo)
+      delta <- Gen.chooseNum(minOrdinalValue, maxOrdinalValue)
+      ordinal = Refined.unsafeApply[Long, NonNegative](delta)
+    } yield (TrustMap.empty.copy(trust = trustInfo), SnapshotOrdinal(ordinal))
+
+    forall(gen) {
+      case (trust, ordinal) =>
+        for {
+          store <- TrustStorage.make(trust)
+
+          maybeOrdinalPublicTrust <- store.updateNext(ordinal)
+
+          expectedOrdinalPublicTrust = trust.toPublicTrust
+          next <- store.getNextOrdinalTrust
+        } yield
+          expect.all(
+            Order.lt(SnapshotOrdinal(ordinalTrustUpdateInterval), ordinal),
+            maybeOrdinalPublicTrust.map(_.ordinal) === SnapshotOrdinal(1000L).some,
+            maybeOrdinalPublicTrust.map(_.labels) === expectedOrdinalPublicTrust.some,
+            trust === next.trust
+          )
+    }
+  }
+
+  test(
+    "current ordinal trust storage doesn't update from next ordinal trust storage because the delay expected ordinal has not been reached"
+  ) {
+    val minOrdinalValue = ordinalTrustUpdateInterval.value
+    val maxOrdinalValue = ordinalTrustUpdateInterval + ordinalTrustUpdateDelay - 1L
+
+    val gen = for {
+      trustInfo <- Gen.mapOfN(4, genPeerTrustInfo)
+      delta <- Gen.chooseNum(minOrdinalValue, maxOrdinalValue)
+      ordinal = Refined.unsafeApply[Long, NonNegative](delta)
+    } yield (TrustMap.empty.copy(trust = trustInfo), SnapshotOrdinal(ordinal))
+
+    forall(gen) {
+      case (trust, ordinal) =>
+        for {
+          store <- TrustStorage.make(trust)
+
+          _ <- store.updateNext(ordinal)
+
+          next <- store.getNextOrdinalTrust
+
+          _ <- store.updateCurrent(ordinal)
+
+          expectedCurrent <- store.getCurrentOrdinalTrust
+        } yield
+          expect.all(
+            Order.gt(SnapshotOrdinal(ordinalTrustUpdateInterval |+| ordinalTrustUpdateDelay), ordinal),
+            trust === next.trust,
+            OrdinalTrustMap.empty === expectedCurrent
+          )
+    }
+  }
+
+  test(
+    "current ordinal trust storage updates from next ordinal trust storage because the delay expected ordinal has been reached"
```

Comment:
as above

Reply:
Changed to `next ordinal trust storage doesn't update from gossip when the gossip ordinal doesn't match the next ordinal`.